healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package healthcheck
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc/internal/backoff"
+)
+
+func TestNewClientHealthCheckBackoff(t *testing.T) {
```

Comment:
With some minor restructuring, you can test multiple backoffs without incurring the sleep times or worrying about the imprecision of time.

I.e.:

```go
var backoffFunc = func(ctx context.Context(), d time.Duration) bool {
	timer := time.NewTimer(d)
	select {
	case <-timer.C:
		return true
	case <-ctx.Done():
		timer.Stop()
		return false
	}
}
```

Now your test can override this variable with a function that captures the delay without sleeping.